app/services/invite_schools.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+class InviteSchools
+  def run(school_ids)
+    logger = Rails.logger
+    logger.info "Emailing schools"
+
+    school_ids.each do |school_id|
+      school = School.find_by(id: school_id)
+
+      if school.nil?
+        logger.info "School not found: #{school_id} ... skipping"
+        next
+      end
+
+      token = reference
+      recipient = recipient_email(school)
+
+      school.nomination_emails.create!(
+        token: token,
+        sent_to: recipient,
+        sent_at: Time.zone.now,
+      )
+      send_nomination_email(recipient, token)
+    rescue StandardError
+      logger.info "Error emailing school: #{school_id} ... skipping"
```

Comment:
We need to record this against the NomintionEmail in the db

Reply:
I doubt it. This covers things like no email recipient being available so the email can't be sent or generally when the nomination email record is invalid. If we want to record those then we need to loosen the validations at the db level